homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,235 @@
+"""Cover entities for the Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from motionblindsble.const import MotionBlindType, MotionRunningType
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityDescription,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_BLIND_TYPE, CONF_MAC_CODE, DOMAIN, ICON_VERTICAL_BLIND
+from .entity import MotionblindsBLEEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class MotionCoverEntityDescription(CoverEntityDescription):
+    """Entity description of a cover entity with default values."""
+
+    key: str = CoverDeviceClass.BLIND.value
+    translation_key: str = CoverDeviceClass.BLIND.value
+
+
+COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
```

Comment:
Might be a bit less to digest here if you made a mapping and than built with with a dict comp

Reply:
I personally think the above is a bit easier to read and understand as opposed to something like this: 

```python
DEVICE_CLASS_MAPPING: dict[CoverDeviceClass, list[MotionBlindType]] = {
    CoverDeviceClass.SHADE: [MotionBlindType.HONEYCOMB, MotionBlindType.ROMAN, MotionBlindType.ROLLER, MotionBlindType.DOUBLE_ROLLER],
    CoverDeviceClass.BLIND: [MotionBlindType.VENETIAN, MotionBlindType.VENETIAN_TILT_ONLY],
    CoverDeviceClass.CURTAIN: [MotionBlindType.CURTAIN, MotionBlindType.VERTICAL],
}

COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
    blind_type.name: MotionCoverEntityDescription(
        device_class=device_class,
        icon=ICON_VERTICAL_BLIND if blind_type == MotionBlindType.VERTICAL else None
    )
    for device_class, blind_types in DEVICE_CLASS_MAPPING.items()
    for blind_type in blind_types
}
```

Reply:
I'd write it like this

```
diff --git a/homeassistant/components/motionblinds_ble/cover.py b/homeassistant/components/motionblinds_ble/cover.py
index 6bb800f5efa..195c13ba606 100644
--- a/homeassistant/components/motionblinds_ble/cover.py
+++ b/homeassistant/components/motionblinds_ble/cover.py
@@ -35,31 +35,28 @@ class MotionCoverEntityDescription(CoverEntityDescription):
     translation_key: str = CoverDeviceClass.BLIND.value
 
 
-COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
-    MotionBlindType.HONEYCOMB.name: MotionCoverEntityDescription(
-        device_class=CoverDeviceClass.SHADE
-    ),
-    MotionBlindType.ROMAN.name: MotionCoverEntityDescription(
-        device_class=CoverDeviceClass.SHADE
-    ),
-    MotionBlindType.ROLLER.name: MotionCoverEntityDescription(
+SHADE_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
     device_class=CoverDeviceClass.SHADE
-    ),
-    MotionBlindType.DOUBLE_ROLLER.name: MotionCoverEntityDescription(
-        device_class=CoverDeviceClass.SHADE
-    ),
-    MotionBlindType.VENETIAN.name: MotionCoverEntityDescription(
-        device_class=CoverDeviceClass.BLIND
-    ),
-    MotionBlindType.VENETIAN_TILT_ONLY.name: MotionCoverEntityDescription(
+)
+BLIND_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
     device_class=CoverDeviceClass.BLIND
-    ),
-    MotionBlindType.CURTAIN.name: MotionCoverEntityDescription(
+)
+CURTAIN_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
     device_class=CoverDeviceClass.CURTAIN
-    ),
-    MotionBlindType.VERTICAL.name: MotionCoverEntityDescription(
+)
+VERTICAL_CURTAIN_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
     device_class=CoverDeviceClass.CURTAIN, icon=ICON_VERTICAL_BLIND
-    ),
+)
+
+COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
+    MotionBlindType.HONEYCOMB.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.ROMAN.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.ROLLER.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.DOUBLE_ROLLER.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.VENETIAN.name: BLIND_ENTITY_DESCRIPTION,
+    MotionBlindType.VENETIAN_TILT_ONLY.name: BLIND_ENTITY_DESCRIPTION,
+    MotionBlindType.CURTAIN.name: CURTAIN_ENTITY_DESCRIPTION,
+    MotionBlindType.VERTICAL.name: VERTICAL_CURTAIN_ENTITY_DESCRIPTION,
 }
 
 ```

Reply:
Makes sense, done thanks!